### PR TITLE
Interfaces and tests

### PR DESCRIPTION
**Pull request by jmcarp**
More tests needed, but this is a start, and I'd be interested in feedback from @afeld and @jcscottiii on what I have so far, since I'm still pretty new to testing in go.

Note--this branches off #10, so just the last two commits are new.

### CODE REVIEW

File: broker/broker.go [hunk 1]
```diff
@@ -60,22 +57,25 @@ func (b *CdnServiceBroker) Provision(
 		return spec, errors.New("must be invoked with `domain` and `origin` keys")
 	}
 
-	_, err = models.NewRoute(b.Settings, b.DB, instanceId, options.Domain, options.Origin)
+	_, err = b.Manager.Create(instanceId, options.Domain, options.Origin)
```
Comment (afeld) [inline broker/broker.go]: Mind using a non-single-letter variable here? Ditto for the rest of the files.

**Comment by afeld** [comment]
Looks good!

File: broker/broker.go (hunk 1 above)
Comment (jmcarp) [inline broker/broker.go]: Here's where the short receiver names come from: https://github.com/golang/go/wiki/CodeReviewComments#receiver-names. Go seems to favor short variable names in general, and very short receiver names in particular.

File: broker/broker_test.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+type ProvisionSuite struct {
+	suite.Suite
+	Broker broker.CdnServiceBroker
+}
+
+func (s *ProvisionSuite) SetupTest() {
+	s.Broker = broker.CdnServiceBroker{
+		Manager: &FakeRouteManager{
+			RouteCreate: models.Route{State: "provisioned"},
```
Comment (afeld) [inline broker/broker_test.go]: Not a blocker, but we might want to try representing the `State` as an enum.

http://stackoverflow.com/questions/14426366/what-is-an-idiomatic-way-of-representing-enums-in-go

Comment (jmcarp) [inline broker/broker_test.go]: Agreed--this is what I did here originally, but combining custom types with `gorm` is a little annoying. `gorm` doesn't know how to marshal and unmarshal custom types, so our `State` types has to implement the `Valuer` and `Scanner` interfaces. Which seemed like a lot of boilerplate for an enum at the time, but yeah, it's probably worth it.

**Pull request merged**